src/journal-remote/journal-gatewayd.c

```diff
@@ -256,7 +256,7 @@ static ssize_t request_reader_entries(
         errno = 0;
         k = fread(buf, 1, n, m->tmp);
         if (k != n) {
-                log_error("Failed to read from file: %s", errno != 0 ? strerror_safe(errno) : "Premature EOF");
+                log_error("Failed to read from file: %s", errno != 0 ? STRERROR(errno) : "Premature EOF");
```

Comment:
(creates the urge for me to add STERROR_OR_EOF() which returns "EOF" in case of zero parameter passed in)